snuba/web/rpc/v1/resolvers/R_eap_spans/common/sampling_in_storage_util.py

```diff
@@ -52,64 +76,96 @@ def _get_most_downsampled_tier() -> Tier:
             ) as tier_specific_span:
-                estimated_query_duration_to_this_tier = (
-                    most_downsampled_query_duration_ms
+                estimated_query_bytes_scanned_to_this_tier = (
+                    most_downsampled_query_bytes_scanned
                     * cast(int, DOWNSAMPLING_TIER_MULTIPLIERS.get(tier))
                 )
 
-                tier_specific_span.set_data(
-                    "estimated_query_duration_to_this_tier",
-                    estimated_query_duration_to_this_tier,
-                )
-                metrics_backend.timing(
-                    "sampling_in_storage_estimated_query_duration",
-                    estimated_query_duration_to_this_tier,
-                    tags={"referrer": referrer, "tier": str(tier)},
+                _record_value_in_span_and_DD(
+                    span,
+                    metrics_backend.distribution,
+                    _SAMPLING_IN_STORAGE_PREFIX + "estimated_query_bytes_scanned",
+                    estimated_query_bytes_scanned_to_this_tier,
+                    {"referrer": referrer, "tier": str(tier)},
                 )
 
-                time_budget = _get_time_budget()
-                if (
-                    estimated_query_duration_to_this_tier
-                    <= time_budget - most_downsampled_query_duration_ms
-                ):
+                bytes_scanned_limit = _get_bytes_scanned_limit()
+                if estimated_query_bytes_scanned_to_this_tier <= bytes_scanned_limit:
                     target_tier = tier
-                    estimated_target_tier_query_duration = (
-                        estimated_query_duration_to_this_tier
+                    estimated_target_tier_bytes_scanned = (
+                        estimated_query_bytes_scanned_to_this_tier
                     )
 
-                tier_specific_span.set_data("target_tier", target_tier)
-                tier_specific_span.set_data("time_budget", time_budget)
+                tier_specific_span.set_data(
+                    _SAMPLING_IN_STORAGE_PREFIX + "target_tier", target_tier
+                )
+                tier_specific_span.set_data(
+                    _SAMPLING_IN_STORAGE_PREFIX + "bytes_scanned_limit",
+                    bytes_scanned_limit,
+                )
 
-        metrics_backend.timing(
-            "sampling_in_storage_routed_tier",
+        timer.mark(_ESTIMATION_END_MARK)
+        _record_value_in_span_and_DD(
+            span,
+            metrics_backend.timing,
+            _SAMPLING_IN_STORAGE_PREFIX + "time_to_run_storage_routing_algo",
+            timer.get_duration_between_marks(
+                _ESTIMATION_START_MARK, _ESTIMATION_END_MARK
+            ),
+            tags={"referrer": referrer, "tier": str(target_tier)},
+        )
+        _record_value_in_span_and_DD(
+            span,
+            metrics_backend.timing,
+            _SAMPLING_IN_STORAGE_PREFIX + "target_tier",
```

Comment:
this should not be a timing metric. What we want is the count of queries routed to each tier